src/js/lib/monster.routing.js

```diff
@@ -104,12 +104,20 @@ define(function(require) {
 		},
 
 		addDefaultRoutes: function() {
-			this.add('apps/{appName}:?query:', function(appName, query) {
+			this.add(/^apps\/([^/?]+)(\/[^?]*)?\??(.*)?$/, function(appName, restSegment, queryString) {
 				// not logged in, do nothing to preserve potentially valid route to load after successful login
 				if (!monster.util.isLoggedIn()) {
 					return;
 				}
-				var availableApps = monster.util.listAppStoreMetadata('user');
+
+				// Do not reload app if it is already loaded and there is a rest segment,
+				// to allow the app to handle its own routing
```

Comment:
How about creating meaningful variable names instead of adding comments here, as comments don't tend to be updated when code changes?

I feel like the following conveys the same information:
```js
var isActiveApp = monster.apps.getActiveApp() === appName,
    hasCustomRouting = !_.isEmpty(restSegment);

```

Reply:
You are right, is easier to read and understand that way 😮